test/integration/acceptance/fake_peer_example_test.cpp

```diff
@@ -40,30 +40,26 @@ class FakePeerExampleFixture : public AcceptanceFixture {
    */
   IntegrationTestFramework &prepareState(size_t num_fake_peers) {
     // request the fake peers construction
-    std::vector<std::future<FakePeerPtr>> fake_peers_futures;
-    std::generate_n(std::back_inserter(fake_peers_futures),
-                    num_fake_peers,
-                    [this] { return itf_->addInitialPeer({}); });
+    itf_->initPipeline(kAdminKeypair);
 
-    itf_->setInitialState(kAdminKeypair);
-
-    auto permissions =
-        interface::RolePermissionSet({Role::kReceive, Role::kTransfer});
-
-    // get the constructed fake peers & set honest behaviour
-    for (auto &fake_peer_future : fake_peers_futures) {
-      assert(fake_peer_future.valid() && "fake peer must be ready");
-      FakePeerPtr fake_peer = fake_peer_future.get();
+    // make the fake peers with honest behaviour
+    std::generate_n(std::back_inserter(fake_peers_), num_fake_peers, [this] {
```

Comment:
I've expected that network generation will be a part of the framework. Maybe move it to ITF?

Reply:
IMHO this will only complicate the ITF interfce. The code that initializes fake peers is only one loop with a single function to get the peer and another to set its behaviour. Current approach keeps things simple and flexible.

Reply:
Yes, the point here is not to change the interface but leave it some shared place. Otherwise, developers will write the loop on each test case.